src/Substrait.Core/ProtoConverter.cs

```diff
@@ -0,0 +1,154 @@
+﻿using Substrait.Expression;
+using Substrait.Protobuf;
+using Substrait.Relation;
+using System.Collections.Immutable;
+using ProtoRel = Substrait.Protobuf.Rel;
+using ProtoRex = Substrait.Protobuf.Expression;
+using Rel = Substrait.Relation.Rel;
+
+namespace Substrait.Core
+{
+  /// <summary>
+  /// Utility to convert Substrait protobuf messages to/from charp domain-objects.
+  /// </summary>
+  public class ProtoConverter
+  {
+    /// <summary>
+    /// Create domain object specific to relational operator type from protobuf message
+    /// </summary>
+    /// <param name="protoRel">Relation encoded as protobuf message</param>
+    /// <returns><see cref="Rel"/> corresponding to message's operator type</returns>
```

Comment:
Minor nit, feel free to overrule.  I feel that using both `operator` and `relation` might be confusing (what is the difference between these two things?).

```suggestion
    /// Create domain object from a protobuf encoded message
    /// </summary>
    /// <param name="protoRel">Relation encoded as protobuf message</param>
    /// <returns><see cref="Rel"/> corresponding to message's relation</returns>
```